File: src/main/java/personal/alexgarland/helloworldmvc/controller/ErrorTestController.java
package personal.alexgarland.helloworldmvc.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
public class ErrorTestController {

  @RequestMapping("/errorTest")
  public ModelAndView getErrorTestPage() {
    return new ModelAndView("errorTest");
  }

  @RequestMapping("/generateServerError")
  public ModelAndView generateServerError() {
    throw new RuntimeException("Something bad happened!");
  }

}

File: src/main/kotlin/personal/alexgarland/helloworldmvc/controller/ErrorTestController.kt
package personal.alexgarland.helloworldmvc.controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView

@Controller
class ErrorTestController {

    val errorTestPage: ModelAndView
        @RequestMapping("/errorTest")
        get() = ModelAndView("errorTest")

    @RequestMapping("/generateServerError")
    fun generateServerError(): ModelAndView {
        throw RuntimeException("Something bad happened!")
    }

}
